### PR TITLE
fix(#1676): app-header menu should close after menu item is clicked

### PR DESCRIPTION
**Pull request by vanessatran-ddi**
Fixing #1676 

https://github.com/GovAlta/ui-components/assets/120135417/3f4d70e3-8305-47fe-a199-6ec6a4a72659

Angular code: 
```
<goa-app-header url="https://example.com" heading="Ticket and Fine Payments">
  <a href="#">Support</a>
  <goa-app-header-menu heading="Tickets" leadingIcon="ticket">
    <a href="#" (click)="selectMenu('Cases'); $event.preventDefault()">Cases</a>
    <a href="#" (click)="selectMenu('Payments'); $event.preventDefault()">Payments</a>
    <a href="#" (click)="selectMenu('Outstanding');$event.preventDefault()">Outstanding</a>
    <a href="#">Test</a>
  </goa-app-header-menu>
  <a href="#" class="interactive">Sign in</a>
</goa-app-header>

<span>Selected Menu: {{selectedMenu}}</span>
```

```
import { Component } from "@angular/core";

@Component({
  selector: "abgov-app-header",
  templateUrl: "./app-header.component.html"
})
export class AppHeaderComponent {
  selectedMenu: string = "";
  constructor() {}
  selectMenu(text: string): void {
    this.selectedMenu = text;
  }
}

```

### CODE REVIEW

**Comment by lizhuomeng71** [comment]
while testing we notice issue is fixed for desktop version, but for mobile (or smaller windows size),  the issue still exist.

**Comment by lizhuomeng71** [comment]
Issue is verified on chrome and firebox for react and angular

File: libs/web-components/src/components/app-header-menu/app-header-menu.spec.ts [hunk 1]
```diff
@@ -65,6 +78,7 @@ describe("Mobile", () => {
   let $: Query;
   let $$: QueryAll;
   let heading: string;
+  let result: any;
```
Comment (chrisolsen) [inline libs/web-components/src/components/app-header-menu/app-header-menu.spec.ts:81]: What is the reason the `result` is moved out of the original scope?

File: libs/web-components/src/components/app-header-menu/AppHeaderMenuWrapper.test.svelte [hunk 2]
```diff
@@ -6,10 +6,19 @@
...
+      target.innerHTML = "Test without loading";
+    }
+  }
 </script>
 
 <GoAAppHeaderMenu {heading} {leadingicon}>
   <a href="#seniors">Seniors</a>
   <a href="#family">Family</a>
   <a href="#children">Children</a>
+  <a href="#special" on:click|preventDefault={selectWithoutLoading}>Test</a>
```
Comment (chrisolsen) [inline libs/web-components/src/components/app-header-menu/AppHeaderMenuWrapper.test.svelte:22]: the `preventDefault` is being removed in Svelte 5, so we should probably avoid using it and move that call into the handler function.

File: libs/web-components/src/components/app-header-menu/app-header-menu.spec.ts [hunk 3]
```diff
@@ -136,10 +150,10 @@ describe("Mobile", () => {
     await user.keyboard("{enter}")
     await waitFor(() => {
       const links = $$("a");
-      expect(links.length).toBe(3);
+      expect(links.length).toBe(4);
     })
 
-    // close
+    // // close
```
Comment (chrisolsen) [inline libs/web-components/src/components/app-header-menu/app-header-menu.spec.ts:156]: probably don't need this.

**Review by chrisolsen — APPROVED** [review]

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**